src/Features/LanguageServer/ProtocolUnitTests/CodeLens/CSharpCodeLensTests.cs

```diff
@@ -227,17 +227,25 @@ void UseM()
         var cache = testLspServer.GetRequiredLspService<CodeLensCache>();
         Assert.NotNull(cache.GetCachedEntry(firstResultId));
 
+        // Update the document so the syntax version changes
+        await testLspServer.OpenDocumentAsync(documentUri);
+        await testLspServer.InsertTextAsync(documentUri, (0, 0, "A"));
+
         // Execute a few more requests to ensure the first request is removed from the cache.
         await testLspServer.ExecuteRequestAsync<LSP.CodeLensParams, LSP.CodeLens[]?>(LSP.Methods.TextDocumentCodeLensName, codeLensParams, CancellationToken.None);
+        await testLspServer.InsertTextAsync(documentUri, (0, 0, "B"));
         await testLspServer.ExecuteRequestAsync<LSP.CodeLensParams, LSP.CodeLens[]?>(LSP.Methods.TextDocumentCodeLensName, codeLensParams, CancellationToken.None);
+
+        await testLspServer.InsertTextAsync(documentUri, (0, 0, "C"));
         var lastCodeLenses = await testLspServer.ExecuteRequestAsync<LSP.CodeLensParams, LSP.CodeLens[]?>(LSP.Methods.TextDocumentCodeLensName, codeLensParams, CancellationToken.None);
         Assert.True(lastCodeLenses.Any());
 
         // Assert that the first result id is no longer in the cache.
         Assert.Null(cache.GetCachedEntry(firstResultId));
 
-        // Assert that the request throws because the item no longer exists in the cache.
-        await Assert.ThrowsAsync<RemoteInvocationException>(async () => await testLspServer.ExecuteRequestAsync<LSP.CodeLens, LSP.CodeLens>(LSP.Methods.CodeLensResolveName, firstCodeLens, CancellationToken.None));
+        // Assert that the items missing from the cache are still fetched
```

Comment:
```suggestion
        // Assert that we get a partially resolved code lens since the resolve request is for a prior document version.
```

As an aside, it might be good to add a test for the case where the syntax version matches but the item doesn't exist in the cache.

Reply:
Good idea!